### PR TITLE
Add story containing inline link to Paragraph

### LINKED ISSUE #721: Add "with inline-link" story to paragraph
**Is your feature request related to a problem? Please describe.**
We should add a "paragraph with inline link" story for more realistic testing of the inline-link component as the inline-link component doesn't have typography styling meaning it's not production equivalent.  The link should be between 2 items of text to give a better example of how the inline link styling affects both the line of text both above and below the link. 

**Describe the solution you'd like**
- Similar to [the Caption story](https://github.com/bbc/psammead/issues/720) we should create a story that places the inline-link between 2 sets of text.

**Describe alternatives you've considered**
We could split the text inputs before and after the inline link to give better control but that would include changing all the translations which might be a future optimisation.

**Testing notes**
Regression test storybook paragraph inline link story

**Additional context**
Raised while working on https://github.com/bbc/psammead/pull/707

### PR DESCRIPTION
**Pull request by pjlee11**
Resolves #721

**Overall change:** _Adds a story for the Paragraph to contain an Inline Link and pulls in @bbc/psammead-inline-link@1.1.0_

**Code changes:**

- Pulls in `@bbc/psammead-inline-link@1.1.0`
- Update package.json, package-lock.json and changelog

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

**Comment by pjlee11** [comment]
blocked on https://github.com/bbc/psammead/issues/716

File: packages/components/psammead-paragraph/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
+| 1.0.6   | [PR#xxx](https://github.com/bbc/psammead/pull/xxx) Use `@bbc/psammead-inline-link@1.1.0` and add a 'with inline link' story |
```
Comment (pjlee11) [inline packages/components/psammead-paragraph/CHANGELOG.md]:
```suggestion
| 1.0.6   | [PR#746](https://github.com/bbc/psammead/pull/746) Use `@bbc/psammead-inline-link@1.1.0` and add a 'with inline link' story |
```

File: packages/components/psammead-paragraph/package.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
     "@bbc/psammead-styles": "^0.3.2"
   },
   "devDependencies": {
+    "@bbc/psammead-inline-link": "1.0.0",
```
Comment (pjlee11) [inline packages/components/psammead-paragraph/package.json]:
```suggestion
    "@bbc/psammead-inline-link": "1.1.0",
```

Bump to `1.1.0` after https://github.com/bbc/psammead/pull/707 is merged and deployed. Also run `npm install`

**Comment by pjlee11** [comment]
image

image

**Review by j-pendlebury — APPROVED** [review]

**Review by ryanmccombe — APPROVED** [review]

**Comment by pjlee11** [comment]
image

image

**Comment by sareh** [comment]
Discussing this with @jamesbrumpton & @nataliesmart: 
When the text for the link is exactly the width of the viewport and you hover/focus, the styling adds padding of 2px on each side and therefore pushes a word down onto the next line. This jump is quite jarring. [1]
However, we discussed how the inline link item is to be used within an article body only - so in a paragraph in the main body copy, or a paragraph within a caption (for an image or video) and in all these scenarios, the components have a minimum of 8px spacing on either side of the text. So this 'jumping' behaviour will not be seen in an Article. [2]
However, we should add a note to the Readme of this component, under usage, to strongly suggest margin or padding on the left and right of 2px or more, whenever using this in production. 

**[1]** Without spacing on left and right:
inline implementation

**[2]** With spacing of 2px on left and right:
inline implementation with 2px spacing left and right

**Comment by jamesbrumpton** [comment]
LGTM

File: packages/components/psammead-paragraph/CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
+| 1.0.7   | [PR#746](https://github.com/bbc/psammead/pull/746) Use `@bbc/psammead-inline-link@1.1.0` and add a 'with inline link' story |
```
Comment (pjlee11) [inline packages/components/psammead-paragraph/CHANGELOG.md]:
```suggestion
| 1.0.7   | [PR#746](https://github.com/bbc/psammead/pull/746) Use `@bbc/psammead-inline-link@1.1.1` and add a 'with inline link' story |
```

**Comment by pjlee11** [comment]
@sareh great spot, I've opened https://github.com/bbc/psammead/pull/775. Going to merge this PR and we can discuss the README in 775

**Comment by sareh** [comment]
@pjlee11 It was all spotted by @jamesbrumpton! I just added the gifs 😄 Fixing forwards sounds good!

**Pull request merged**